pallet/src/migrations.rs

```diff
@@ -0,0 +1,80 @@
+use super::*;
+
+use frame_support::{
+    dispatch::GetStorageVersion, pallet_prelude::StorageVersion, traits::Get, weights::Weight,
+};
+use sp_std::vec::Vec;
+use structs::*;
+
+pub(crate) fn on_runtime_upgrade<T: Config>() -> Weight {
+    v0::MigrateToV0x::<T>::on_runtime_upgrade()
+}
+
+const CURRENT_STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+pub mod v0 {
+
+    use super::*;
+
+    // This migration sort all values in some StorageMaps allowing them to be binary searchable
+    // As there was no storage version set before, we set it now to 0
+    pub struct MigrateToV0x<T>(sp_std::marker::PhantomData<T>);
+
+    impl<T: Config> MigrateToV0x<T> {
+        pub fn on_runtime_upgrade() -> Weight {
+            let current_storage_version: StorageVersion = Pallet::<T>::current_storage_version();
+
+            if current_storage_version.eq(&CURRENT_STORAGE_VERSION) {
```

Comment:
Does it mean we'll enter this migration part again if we have the next runtime upgrade? Because the code write CURRENT_STORAGE_VERSION 0 back again

You can also check this migration implementation. https://github.com/peaqnetwork/peaq-pallet-mor/commit/43401a3dcb4a7a270899034efd66633e27f7bd8f

Reply:
Good point, lemme look into that.

Reply:
Migration implemented and fixed, just adding weights now